test/integration/full/preload-cssom/preload-cssom.js

```diff
@@ -0,0 +1,202 @@
+/* global axe, Promise */
+describe('preload cssom integration test pass', function() {
+	'use strict';
+
+	var origAxios;
+
+	before(function(done) {
+		function start() {
+			// Stop messing with my tests Mocha!
+			document.querySelector('#mocha h1').outerHTML =
+				'<h2>preload cssom integration test</h2>';
+			// cache original axios object
+			if (axe.imports.axios) {
+				origAxios = axe.imports.axios;
+			}
+			// run axe
+			axe.run(
+				{
+					preload: true
+				},
+				function(err) {
+					assert.isNull(err);
+					done();
+				}
+			);
+		}
+		if (document.readyState !== 'complete') {
+			window.addEventListener('load', start);
+		} else {
+			start();
+		}
+	});
+
+	function createStub(shouldReject) {
+		axe.imports.axios = function stubbedAxios() {
+			return new Promise(function(resolve, reject) {
+				if (shouldReject) {
+					reject(new Error('Fake Error'));
+				}
+				resolve({
+					data: 'body{overflow:auto;}'
+				});
+			});
+		};
+	}
+
+	function restoreStub() {
+		if (origAxios) {
+			axe.imports.axios = origAxios;
+		}
+	}
+
+	function assertStylesheet(sheet, selectorText, cssText) {
+		assert.isDefined(sheet);
+		assert.property(sheet, 'cssRules');
+		assert.equal(sheet.cssRules[0].selectorText, selectorText);
+		assert.equal(sheet.cssRules[0].cssText.replace(/\s/g, ''), cssText);
+	}
+
+	function getPreload(root) {
+		var config = {
+			asset: 'cssom',
+			timeout: 10000,
+			treeRoot: axe.utils.getFlattenedTree(root ? root : document)
+		};
+		return axe.utils.preloadCssom(config);
+	}
+
+	function commonTestsForRootAndFrame(root) {
+		it('should return external stylesheet from cross-domain and verify response', function(done) {
+			getPreload(root).then(function(results) {
+				var sheets = results[0];
+				var externalSheet = sheets.filter(function(s) {
+					return s.isExternal;
+				})[0];
+				assertStylesheet(externalSheet, 'body', 'body{overflow:auto;}');
+				done();
+			});
+		});
+
+		it('should reject external stylesheets', function(done) {
+			restoreStub();
+			createStub(true);
+			var doneCalled = false;
+			getPreload(root).catch(function(error) {
+				assert.equal(error.message, 'Fake Error');
+				if (!doneCalled) {
+					done();
+				}
+				doneCalled = true;
+			});
+		});
+	}
+
+	beforeEach(function() {
+		if (window.PHANTOMJS) {
+			assert.ok('PhantomJS is a liar');
```

Comment:
Haha, yes it is, but this assertion should probably just be a comment instead.

Reply:
Nah, keeping this. I see this as a part of some old code :)

Reply:
Oh well. No harm

Reply:
Once we remove Phantom those can go. But until then it was a workaround for features lacking in Phantom or requiring painful workarounds